api/src/main/java/jakarta/servlet/ServletOutputStream.java

```diff
@@ -51,6 +53,37 @@ public abstract class ServletOutputStream extends OutputStream {
     protected ServletOutputStream() {
     }
 
+    /**
+     * Writes from the given buffer to the output stream. The method blocks until all the remaining data in the buffer has
+     * been written.
+     * <p>
+     * If this method is called when the buffer has no data remaining, the method returns immediately.
+     * <p>
+     * When the method returns, and if data has been written, the buffer's limit will be unchanged from the value when
+     * passed to this method and the position will be the same as the limit.
+     * <p>
+     * Subclasses are strongly encouraged to override this method and provide a more efficient implementation.
+     *
+     * @param buffer The buffer from which the data is written
+     *
+     * @exception IOException If the output stream has been closed or if some other I/O error occurs.
+     *
+     * @exception NullPointerException If buffer is null
+     */
+    public void write(ByteBuffer buffer) throws IOException {
+        Objects.nonNull(buffer);
```

Comment:
I think you wanted to use `requireNonNull()` instead.

Reply:
That would be rather more useful wouldn't it ;).
Tx for spotting that. I'll get it fixed shortly.